### PR TITLE
Bundlerevisionauto

### PR DESCRIPTION
**Pull request by NickDickinsonWilde**
Allow to automatically select new revision for media entities per bundle.

### CODE REVIEW

File: src/Entity/MediaBundle.php [hunk 1]
```diff
@@ -174,4 +182,18 @@ protected function typePluginCollection() {
     return $this->typePluginCollection;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function isNewRevision() {
```
Comment (slashrsm) [inline src/Entity/MediaBundle.php:188]: This and function below are not defined by any interface that MediaBundle implements. They should be added to MediaBundleInterface.

File: src/MediaBundleForm.php [hunk 2]
```diff
@@ -209,6 +215,7 @@ protected function copyFormValuesToEntity(EntityInterface $entity, array $form,
   public function save(array $form, FormStateInterface $form_state) {
     /** @var  \Drupal\media_entity\MediaBundleInterface $bundle */
     $bundle = $this->entity;
+    $bundle->setNewRevision($form_state->getValue('new_revision'));
```
Comment (slashrsm) [inline src/MediaBundleForm.php:219]: Is this respected when saving an actual media entity?

Comment (NickDickinsonWilde) [inline src/MediaBundleForm.php:219]: Yes, it is respected.

File: src/Entity/MediaBundle.php (hunk 1 above)
Comment (NickDickinsonWilde) [inline src/Entity/MediaBundle.php:188]: oops, neglected to commit the interface changes, added commit to pull req.

File: src/MediaBundleForm.php [hunk 3]
```diff
@@ -154,6 +154,12 @@ public function form(array $form, FormStateInterface $form_state) {
       );
     }
 
+    $form['new_revision'] = [
```
Comment (slashrsm) [inline src/MediaBundleForm.php]: Add #description and move it up (below description field).

File: src/MediaBundleForm.php (hunk 2 above)
Comment (slashrsm) [inline src/MediaBundleForm.php:219]: I tested this manually and it doesn't seem to be. Steps:
- create media bundle (check "Create new revision")
- create new media entity of the bundle I just created ("Create new revision" checkbox is checked as new entity is being created)
- Navigate back to edit page for the entity I just created
- **Expected result:** "Create new revision" is checked by default
- **Actual result:** Checkbox is not checked

Am I missing something?

File: src/MediaBundleForm.php (hunk 3 above)
Comment (NickDickinsonWilde) [inline src/MediaBundleForm.php]: done :)

File: src/MediaBundleForm.php (hunk 2 above)
Comment (NickDickinsonWilde) [inline src/MediaBundleForm.php:219]: not exactly.
Had to do another change for it to work on _edit_ forms.
Fixed.

**Comment by Berdir** [comment]
See also https://github.com/fago/entity/pull/27, where we're trying to add that by default to entity.module.

**Comment by slashrsm** [comment]
Adopting that would make a lot of sense.

File: src/MediaBundleForm.php (hunk 2 above)
Comment (ericduran) [inline src/MediaBundleForm.php:219]: This line shouldn't be needed. `copyFormValuesToEntity` should handled that in `parent::copyFormValuesToEntity($entity, $form, $form_state);`

File: src/MediaBundleInterface.php [hunk 4]
```diff
@@ -67,4 +67,22 @@ public function getTypeConfiguration();
...
    */
   public function setTypeConfiguration($configuration);
+
+  /**
+   * Gets whether a new revision should be created by default.
+   *
+   * @return bool
+   *   TRUE if a new revision should be created by default.
+   */
+  public function isNewRevision();
```
Comment (ericduran) [inline src/MediaBundleInterface.php:77]: I know this comes from core but this is a pretty bad name.

**Comment by slashrsm** [comment]
Improved this patch a bit in https://www.drupal.org/node/2716737#comment-11143309.

**Comment by slashrsm** [comment]
Commited a patch that was based on this branch: 4b9542b

Thank you all!

**Pull request closed without merging**